batch/batch/driver/main.py

```diff
@@ -273,13 +273,37 @@ async def deactivate_instance_1(instance):
     log.info(f'deactivating {instance}')
     await instance.deactivate('deactivated')
     await instance.mark_healthy()
-    return web.Response()
 
 
 @routes.post('/api/v1alpha/instances/deactivate')
 @active_instances_only
 async def deactivate_instance(request, instance):  # pylint: disable=unused-argument
-    return await asyncio.shield(deactivate_instance_1(instance))
+    await asyncio.shield(deactivate_instance_1(instance))
+    return web.Response()
+
+
+@routes.post('/instances/{instance_name}/deactivate')
+@check_csrf_token
+@web_authenticated_developers_only()
+async def kill_instance(request, userdata):
+    instance_name = request.match_info['instance_name']
+
+    inst_coll_manager: InstanceCollectionManager = request.app['driver'].inst_coll_manager
+    instance = inst_coll_manager.get_instance(instance_name)
+
+    if instance:
+        pool_name = instance.inst_coll.name
+        pool_url_path = f'/inst_coll/pool/{pool_name}'
+        session = await aiohttp_session.get_session(request)
+
+        if instance.state == 'active':
+            await asyncio.shield(instance.kill())
+            set_message(session, f'Deactivated instance {instance_name}', 'info')
+        else:
+            set_message(session, 'Cannot deactivate a non-active instance', 'error')
+    else:
+        return web.HTTPNotFound()
+    return web.HTTPFound(deploy_config.external_url('batch-driver', pool_url_path))
```

Comment:
The logic here for the two code paths is unnecessarily interleaved, i.e. I have to skip over lines 304-305 in the case where the instance is not None. I'd prefer we make this as continuous as possible, so move this return into the if block, and then we don't need an `else`, our default return will just be `HTTPNotFound`. Alternatively, we could avoid a large block of indentation by flipping this if around and then having the bulk of the code indented at the normal level:

```python
if instance is None:
    return web.HTTPNotFound()
    
… rest of the function …
```

What do you think?